Add uhdm-integration conda package

tjurtsch opened the pull request:
This PR adds Conda package with binaries built from uhdm-integration:
https://github.com/alainmarcel/uhdm-integration

There are two output packages:
- Surelog-UHDM-Verilator flow (uhdm-integration-verilator)
- Surelog-UHDM-Yosys flow (uhdm-integration-yosys)

Commit:
Add uhdm-integration conda package

Signed-off-by: Tomasz Jurtsch <tjurtsch@antmicro.com>

tgorochowik commented on uhdm-integration/meta.yaml:
```diff
@@ -0,0 +1,49 @@
+      script: yosys-flow.sh
+      test:
+          commands:
+              - uhdm-surelog --version
+              - uhdm-yosys --version
```
Could we create separate packages for yosys and verilator? For now maybe it makes sense to keep them together, but in the long run it makes sense for them not to be packed together.

Additionally it will be much easier to integrate them with sv-tests if they are separated.

tgorochowik replied:
Ok, nvm, I wasn't aware you can generate two packages from a single build scripts like you do here!

tjurtsch commented:
I added simple test to check if resulting binaries work and requirements are correct (print tool version).
CI build passes, IMO this PR can be merged.

tjurtsch commented:
@tgorochowik PTAL

tgorochowik reviewed (approved):
LGTM.

@mithro would you mind taking a look too?

mithro commented on .travis.yml:
```diff
@@ -43,6 +43,7 @@ env:
   - PACKAGE=surelog
   - PACKAGE=sv-parser
   - PACKAGE=verible             USE_SYSTEM_GCC_VERSION="8"
+  - PACKAGE=uhdm-integration
```
This should probably be a couple of separate packages;
 * `yosys-udhm`
 * `verilator-uhdm`
 * `surelog-parse`

tgorochowik replied:
@mithro we actually generate two packages from this (there are two outputs defined for `uhdm-yosys` and `uhdm-verilator`) - it probably doesn't make sense to split that in travis as a major part for both outputs is common (namely `surelog` + `uhdm`).

Both packages come with `surelog` inside as there are several reasons to keep it this way:
* The development is quite rapid, having them packed together allows us to be sure that the versions are compatible (`surelog` used here is actually vanilla `surelog` but from a submodule that should be fully compatible with the rest of the integration)
* The way `sv-tests` travis jobs are currently implemented is they uncomment a dependency and download a single conda package depending on the name of the job (I think you are the author of this mechanism) - if we pack `surelog` with `yosys`/`verilator` we won't have to change that mechanism - we probably could try to work around this by defining some dependencies in conda, but the way it is done now just makes things simpler
* In the long run we may want to consider using `surelog` and `udhm` as libraries and compile them into `verilator`/`yosys` to avoid that extra step of dumping `udhm` output into file

mithro replied:
> @mithro we actually generate two packages from this (there are two outputs defined for `uhdm-yosys` and `uhdm-verilator`) - it probably doesn't make sense to split that in travis as a major part for both outputs is common (namely `surelog` + `uhdm`).

You can generate three packages and have the versions locked together?

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#outputs-section

> 
> Both packages come with `surelog` inside as there are several reasons to keep it this way:
> 
> * The development is quite rapid, having them packed together allows us to be sure that the versions are compatible (`surelog` used here is actually vanilla `surelog` but from a submodule that should be fully compatible with the rest of the integration)

Yeap.

> * The way `sv-tests` travis jobs are currently implemented is they uncomment a dependency and download a single conda package depending on the name of the job (I think you are the author of this mechanism) - if we pack `surelog` with `yosys`/`verilator` we won't have to change that mechanism - we probably could try to work around this by defining some dependencies in conda, but the way it is done now just makes things simpler

Dependencies would be a better solution.

> * In the long run we may want to consider using `surelog` and `udhm` as libraries and compile them into `verilator`/`yosys` to avoid that extra step of dumping `udhm` output into file

Hence why I was thinking the modules should be `verilator-uhdm` rather than `uhdm-verilator`?

tgorochowik committed:
uhdm-integration: generate three separate packages

For surelog-uhdm, verilator-uhdm and yosys-uhdm.

Signed-off-by: Tomasz Gorochowik <tgorochowik@antmicro.com>

tgorochowik committed:
uhdm-integration: add meta-packages with dependencies only

This way we can install only one package and be able to use the
whole flow with tools in compatible versions, e.g:
 * Install the uhdm-integration-verilator package
 * Use the surelog->uhdm->verilator flow

Signed-off-by: Tomasz Gorochowik <tgorochowik@antmicro.com>

tgorochowik commented:
@tjurtsch will be out of office for a while so I took over - @mithro could you please take a look again? I've renamed the packages, moved surelog to a separate package and added empty meta-packages which pack surelog+verilator and surelog+yosys as dependencies.

mithro commented:
@tgorochowik Can you resolve the open comment if it has been fixed?

mithro commented on uhdm-integration/meta.yaml:
```diff
@@ -0,0 +1,73 @@
+
+source:
+  git_rev: master
+  git_url: https://github.com/antmicro/uhdm-integration.git
```
Should we be using https://github.com/alainmarcel/uhdm-integration ?

mithro reviewed (approved):
Resolve the open comments then feel free to merge.

tgorochowik committed:
uhdm-integration: use the mainline repo

Signed-off-by: Tomasz Gorochowik <tgorochowik@antmicro.com>
